core/amber/src/main/scala/edu/uci/ics/texera/web/resource/ResultResource.scala

```diff
@@ -21,22 +23,70 @@ class ResultResource extends LazyLogging {
       request: ResultExportRequest,
       @Auth user: SessionUser
   ): Response = {
-
     try {
-      val resultExportService = new ResultExportService(WorkflowIdentity(request.workflowId))
+      request.destination match {
+        case "local" =>
+          // CASE A: multiple operators => produce ZIP
+          if (request.operatorIds.size > 1) {
+            val resultExportService = new ResultExportService(WorkflowIdentity(request.workflowId))
+            val (zipStream, zipFileNameOpt) =
+              resultExportService.exportOperatorsAsZip(user.user, request)
+
+            if (zipStream == null) {
+              return Response
```

Comment:
maybe throw an error here instead of return a Response, because in your catch you will return Response Internal error